activerecord/lib/active_record/associations/preloader/through_association.rb

```diff
@@ -44,15 +42,15 @@ def records_by_owner
 
         private
           def source_preloaders
-            @source_preloaders ||= PRELOADER.preload(middle_records, source_reflection.name, scope)
+            @source_preloaders ||= ActiveRecord::Associations::Preloader.new(middle_records, source_reflection.name, scope, false).preload
```

Comment:
is there any chance to get back to keep `associate_by_default` as a keyword argument? it kind of difficult to follow this boolean argument. It would be the same for `polymorphic_parent = false`

Reply:
I had them all as keyword arguments in a commit I didn't push but it felt unruly. I'll think about it tonight and decide which way to go tomorrow.

Reply:
Thanks!